Source/WebCore/style/InlineTextBoxStyle.cpp

```diff
@@ -181,6 +181,26 @@ static float computedUnderlineOffset(const UnderlineOffsetArguments& context)
         computedUnderlineOffset = std::max<float>(desiredOffset, fontMetrics.ascent());
         break;
     }
+    case TextUnderlinePosition::Left: {
+        if (!context.lineStyle.isHorizontalWritingMode()) {
+            ASSERT(context.textUnderlinePositionUnder);
+            // Position underline relative to the bottom edge of the lowest element's content box.
+            auto desiredOffset = context.textUnderlinePositionUnder->textRunLogicalHeight + gap + std::max(context.textUnderlinePositionUnder->textRunOffsetFromBottomMost, 0.f) + underlineOffset.lengthOr(0.f);
+            computedUnderlineOffset = std::max<float>(desiredOffset, fontMetrics.ascent());
+        } else
+            computedUnderlineOffset = fontMetrics.ascent() + underlineOffset.lengthOr(gap);
+        break;
+    }
+        case TextUnderlinePosition::Right: {
+        if (!context.lineStyle.isHorizontalWritingMode()) {
+            ASSERT(context.textUnderlinePositionUnder);
+            // Position underline relative to the bottom edge of the lowest element's content box.
+            auto desiredOffset = context.textUnderlinePositionUnder->textRunLogicalHeight + gap + std::max(context.textUnderlinePositionUnder->textRunOffsetFromBottomMost, 0.f) + underlineOffset.lengthOr(0.f);
+            computedUnderlineOffset = std::max<float>(desiredOffset, fontMetrics.ascent());
+        } else
+            computedUnderlineOffset = fontMetrics.ascent() + underlineOffset.lengthOr(gap);
+        break;
+    }
```

Comment:
Seems suspicious to me that left/right have the same implementation.

Reply:
They should be combined in the latest patch.